host/sgx/sgxload.c

```diff
@@ -180,36 +180,24 @@ static void* _allocate_enclave_memory(size_t enclave_size, int fd)
         int mflags = MAP_SHARED;
         uint64_t mmap_size = enclave_size;
 
-        /* If no file descriptor, then perform anonymous mapping and double
-         * the allocation size, so that BASE can be aligned on the SIZE
-         * boundary. This isn't neccessary on hardware backed enclaves, since
-         * the driver will do the alignment. */
-        if (fd == -1)
+        /* Perform an anonymous mapping and double the allocation size,
+         * so that BASE can be aligned on the SIZE boundary. This isn't
+         * necessary on hardware backed enclaves, since the driver will
+         * do the alignment. */
+        mflags |= MAP_ANONYMOUS | MAP_NORESERVE;
+        if (oe_safe_mul_u64(mmap_size, 2, &mmap_size) != OE_OK)
```

Comment:
In simulation mode do we still need this logic for alignment on SIZE boundary?
The Windows branch of the code does have alignment this alignment logic.

Reply:
@anakrish I think it was only for verisimilitude of the simulation. Removing it doesn't cause any issues, so I've simplified this down to an mmap of the required size as per the Windows implementation for simplicity.